### DISCUSSION TITLE
Dgus Display wiring ?

### DISCUSSION THREAD

**Discussion by MichalVilimek** [seho85/klipper-dgus#65] [Q&A]
Do I understand correctly that the original display is connected directly to the Raspberry Pi running Klipper?Can anyone advise me how to connect the display to the Raspberry Pi?
Thank you very much
MIchal

**Comment by seho85** [comment]
Please have a look at the documentation that can be found on readthedocs.

Especially this page should answer your question:

https://klipper-dgus.readthedocs.io/en/latest/usage/installation/display_connection.html

**Comment by MichalVilimek** [comment]
And please, is it so that the original display is connected to the Raspberry Pi Gpio or as a second option that I connect the display via a converter to the Raspberry Pi USB?

**Reply by seho85** [reply]
The Display is controlled by an serial interface. 

So you can use either an USB-TTL (USB Serial Interface) **or** you can use the native serial interface available on the pinheader of the Raspberry PI.

The voltage used for data transmission (on RX and TX Pin) need to be 3.3V.

I've heard from some that have had problems using the USB-TTL, but from what we found  out about the problems, it was always a problem of the USB-TTL hardware. They reported the display became unresponsive. After they switched to the serial interface on the pinheader problem was gone. I personally use an USB-TTL, and don't have any problems.

**Reply by MichalVilimek** [reply]
Which USB TTL are you using please ?Would you have a link from where you ordered it ?Thanks

**Reply by seho85** [reply]
I ordered it a few years ago, so i haven't any clue where its coming from. It may come from Aliexpress, but i just can't remind.

It's just a a cloned PL2303.

I would recommend to buy at a local seller, so you can request for an exchange if yours provide any problems.
